Add prompt format as config and CLI option.

j-bennet opened the pull request:
Configurable prompt for mssql-cli. I set the default to `"\\d> "` to mimic the current behavior (even though pgcli and mycli have longer default prompts, e.g. `"\\u@\\h:\\d> "`.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/dbcli/mssql-cli?pullRequest=137) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/dbcli/mssql-cli?pullRequest=137) before we can accept your contribution.<br/><br/>:x: j-bennet [sign now](https://cla.opensource.microsoft.com/dbcli/mssql-cli?pullRequest=137)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/dbcli/mssql-cli?pullRequest=137) it.</sub>

pensivebrian commented on mssqlcli/main.py:
```diff
@@ -98,6 +98,7 @@
 class MssqlCli(object):
 
     max_len_prompt = 30
+    default_prompt = '\\d> '
```
I don't think a trailing space is a common convention. Can we remove the space after the '>' for the default?

j-bennet replied:
I think it is. Both mycli and pgcli use trailing ` ` in their prompt:

https://github.com/dbcli/pgcli/blob/327f15afc3fcbd556b87a5b568b2b6c983519bdd/pgcli/pgclirc#L123

https://github.com/dbcli/mycli/blob/d6ccd390a9b4b659a2bd35e8bf94172eb9c3874e/mycli/myclirc#L67

Without the ` `, once you start typing, it looks like the prompt and the query are glued together:

```
foo>select 1
```

not pretty!

```
foo> select 1
```

pretty!

pensivebrian replied:
Fair enough. I don't feel strongly either way, so a space it is.

pensivebrian commented on mssqlcli/main.py:
```diff
@@ -698,9 +706,10 @@ def get_last_query(self):
 @click.option('--enable-sqltoolsservice-logging', is_flag=True,
               default=False,
               help='Enables diagnostic logging for the SqlToolsService.')
+@click.option('--prompt', help='Prompt format (Default: "\\d> ").')
```
Can we reference the `default_prompt` constant above?

pensivebrian commented on mssqlcli/mssqlcliclient.py:
```diff
@@ -24,7 +24,13 @@ def __init__(self, sql_tools_client, server_name, user_name, password,
+
         self.server_name = server_name
+        if ',' in server_name:
+            self.host, self.port = self.server_name.split(',')
```
since host and port are for prompt display purposes, can you rename them to prompt_host and prompt_port? This way, it's clear we're not using the values when connecting to the target database.

pensivebrian commented on mssqlcli/mssqlclirc:
```diff
@@ -103,6 +103,15 @@ row_limit = 1000
+# \d - Database name
+# \p - Database port
+# \n - Newline
+prompt = '\d> '
```
Same as above, let's we remove the space after the '>' for the default.

pensivebrian commented on mssqlcli/main.py:
```diff
@@ -648,9 +651,14 @@ def get_completions(self, text, cursor_positition):
         # mssql-cli
-        string = self.mssqlcliclient_query_execution.database + u'>'
+        string = string.replace('\\t', self.now.strftime('%x %X'))
+        string = string.replace('\\u', self.mssqlcliclient_query_execution.user_name or '(none)')
```
If the `-E` options is specified, we use windows/kerberos, which defaults to the current user. This is really common for Windows environments. Can we detect if -E was passed, and get the current user via `os.getlogin()`?

pensivebrian replied:
If you don't have a Windows environment to test this, feel free to skip this, and someone with a Windows machine can fix this in another PR.

pensivebrian commented on mssqlcli/mssqlcliclient.py:
```diff
@@ -24,7 +24,13 @@ def __init__(self, sql_tools_client, server_name, user_name, password,
+            self.host, self.port = self.server_name.split(',')
+        else:
+            self.host = server_name
+            self.port = 1433
```
For practical purposes, 1433 is correct. However, SQL Server supports other transports, named pipes and shared memory, which use different ports. Also, the dedicated admin connection uses 1434. However, we can ignore this and fix if we ever get any complaints. I doubt we will :)

pensivebrian commented on mssqlcli/mssqlclirc:
```diff
@@ -103,6 +103,15 @@ row_limit = 1000
 # Skip intro on startup and goodbye on exit
 less_chatty = False
 
+# mysql-cli prompt
```
mssql-cli :)

j-bennet replied:
Yiiikes. :) Good catch.

j-bennet commented:
@pensivebrian I addressed all the concerns, except one - I'd still prefer to keep the trailing space in prompt.

Would you mind trying out the `os.getlogin()` change on windows? Indeed, I do not have access to a Windows machine.

Thank you!

pensivebrian commented:
@j-bennet I led you astray with the `os.getlogin()` suggestion. Turns out, this function doesn't exist on Windows. It appears `getpass.getuser()` from the [getpass ](https://docs.python.org/2.7/library/getpass.html) module is the proper way to get the current user for Windows and Linux.

If you push the getpass change, I'll give it another go on Windows. My apologies for the bad advice.

j-bennet commented:
No problem - changed to `getpass.getuser()`.

pensivebrian commented:
Thanks! Worked perfectly on Windows.

pensivebrian commented:
Paging @abhisheksinha89 and @MrMeemus - I plan on merging this tomorrow morning. If you can, please take a look. Thanks!
